common/app/model/LiveBlogCurrentPage.scala

```diff
@@ -36,9 +36,12 @@ object LiveBlogCurrentPage {
       blocks: Blocks,
       sinceBlockId: SinceBlockId,
       filterKeyEvents: Boolean,
+      topMentionResult: Option[TopMentionsResult],
   ): Option[LiveBlogCurrentPage] = {
+
     val bodyBlocks = blocks.requestedBodyBlocks.get(sinceBlockId.around).toSeq.flatMap { bodyBlocks =>
-      applyFilters(bodyBlocks, filterKeyEvents, None).takeWhile(_.id != sinceBlockId.lastUpdate)
+      val onlyBlocksAfterLastUpdated = bodyBlocks.takeWhile(_.id != sinceBlockId.lastUpdate)
+      applyFilters(onlyBlocksAfterLastUpdated, filterKeyEvents, topMentionResult)
     }
```

Comment:
Here, I changed the order by which we are filtering. The existing code was first filtering, then taking the blocks after the last updated. But this could result in blocks that are also before last updated. For example, 

Imagine we have the following blocks: `1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14`
Imagine filtering them by topics or by key events would give us this:  `5, 6, 7, 11`
Then we want to take the blocks from the filtered list after `9` (last updated is 9)
CAPI would return  `4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14` for `body:around:9:5`
The result of the function will be =>  5, 6, 7, 11 
But we would expect the result to be 11 (11 is the only topic block after 9)

This change is fixing this.

Reply:
nice catch!

Reply:
yep makes sense!